tasks/backfill_commit_data_to_storage.py

```diff
@@ -0,0 +1,153 @@
+        # We can leave to BaseCodeovTask to commit the changes to DB
+        return {
+            "success": report_json_backfill_result["success"]
+            and report_classes_backfill_result["success"],
+            "errors": report_classes_backfill_result.get("errors")
+            + report_json_backfill_result.get("errors"),
+        }
+
+    def handle_report_json(self, dbsession: Session, commit: Commit) -> BackfillResult:
+        if commit._report_json_storage_path is not None:
+            if commit._report_json is not None:
+                log.warning(
+                    "Both _report_json AND _report_json_storage_path are set. Leaving as is.",
+                    extra=dict(commitid=commit.id, commit=commit.commitid),
+                )
+            log.debug(
+                "Commit info already in storage",
+                extra=dict(commitid=commit.id, commit=commit.commitid),
+            )
+            return {"success": True, "errors": []}
+        if commit._report_json is not None:
+            # write to storage and clears out db field
+            commit.report_json = commit._report_json
+            return {"success": True, "errors": []}
+        log.warning(
+            "Neither _report_json nor _report_json_storage_path are set. Nothing to do.",
+            extra=dict(commitid=commit.id, commit=commit.commitid),
+        )
+        return {"success": False, "errors": [BackfillError.missing_data.value]}
+
+    def handle_all_report_rows(
+        self, db_session: Session, commit: Commit
+    ) -> BackfillResult:
+        report_rows = (
+            db_session.query(CommitReport).filter_by(commit_id=commit.id_).all()
+        )
+        if report_rows == []:
+            new_report_row = CommitReport(commit_id=commit.id_, code=None)
+            db_session.add(new_report_row)
+            db_session.flush()
+            report_rows = [new_report_row]
+        aggregate_results = dict(success=True, errors=[])
+        for row in report_rows:
+            result = self.handle_single_report_row(db_session, commit, row)
+            aggregate_results["success"] = (
+                aggregate_results["success"] and result["success"]
+            )
+            aggregate_results["errors"].extend(result["errors"])
+        return aggregate_results
+
+    def handle_single_report_row(
+        self, db_session: Session, commit: Commit, report_row: CommitReport
+    ) -> BackfillResult:
+        report_details = (
+            db_session.query(ReportDetails).filter_by(report_id=report_row.id_).first()
+        )
+        if report_details is None:
+            report_details = ReportDetails(
+                report_id=report_row.id_,
+                _files_array=[],
+                report=report_row,
+            )
+            db_session.add(report_details)
+            db_session.flush()
+
+            repo_yaml = get_repo_yaml(commit.repository)
```

Comment:
Should this be `get_current_yaml` instead so that it takes into account the commit's YAML?

Reply:
I don't think it matters for the operations we are doing. And using the repo_yaml saves us a request to the git provider.

However from a correctness pov - given we will be working mostly with old commits - I guess the most accurate would be to use *only* the commit yaml... (given that more recent changes are probably merged to the repo yaml already and might have been done in the owner yaml)

What do you think is best?

Reply:
I think you're right that it doesn't matter.  Looks like there are 2 methods being called here:

* `report_service.get_existing_report_for_commit_from_legacy_data`
* `report_service.save_report`

I just looked through both and AFAICT neither rely on `self.current_yaml` in any way.  You could probably even pass `ReportService({})` here.